mllib/src/main/scala/org/apache/spark/ml/classification/RandomForestClassifier.scala

```diff
@@ -134,13 +135,23 @@ final class RandomForestClassificationModel private[ml] (
 
   override def treeWeights: Array[Double] = _treeWeights
 
+  override def transform(dataset: DataFrame): DataFrame = {
+    val bcastModel = dataset.sqlContext.sparkContext.broadcast(this)
```

Comment:
You mentioned that we might want to selectively broadcast the model, only if it's large enough.  Do you think that is something we can do here automatically, or would it need to be a configuration setting?